### ISSUE TITLE
Implement a lighter `Solution` struct 

### ISSUE THREAD

**Issue by mvanzulli** [mvanzulli/ONSAS.jl#429]
Since now the `SolutionStates` stores eth form the `StaticState` this is not desirable. Here we have two possibilities, implement a new `Struct` <: `AbstractSolution` and add a `save` while solving the analysis to store the desired fields or a customization  for the user. 

I think the last one is the right path to take. The interface could work like this:

```julia
fields_to_store =[:displacement, :strain, :stress]

options = Options(;fields_to_store)

solution = solve(analysis, algorithm; options)
```

The solve function can be extended
```julia
function solve(analysis::A, algorithm::SS, options::Options=Options([])) 
```

The options struct can start being:

```julia
struct Options
    fields_to_store::Vector{Symbol}
end
```
The solve function can work:
```julia
function solve(analysis::A, algorithm::SS, options::Options=Options([])) 
    where {A<:AbstractAnalysis, SS<:AbstractSolver}
    
    # Initialize the solution object, passing options.fields_to_store if it exists
    solution = StatesSolution(analysis, options.fields_to_store)

    # Perform the analysis 
    while !is_converged(state)
        state_converged = iterate_solver(algorithm, state)

        # Store only the fields the user has specified
        for field in solution.options.fields_to_store
            solution.states[step].field .= getfield(state, field)
        end
    end

    return solution
end
```

**Comment by mvanzulli** [comment]
Discussion takes from the meeting with  @jorgepz and @mforets: 

- Implement a general `OutputState` with all the fields possible across multiple `StructuralState`. This struct might inherit from the `AbstractStructuralState` interface. 
- Check the  `AbstractStructuralState` abstract methods to DRY. 
-  Create a `store!` function to fill the `OutputState` with the `StructuralState`.  The methods are defined in the [Solutions.jl](https://github.com/ONSAS/ONSAS.jl/blob/main/src/StructuralSolvers/Solutions.jl) file.

**Comment by mvanzulli** [comment]
Closed via #474
